iso-mirror.1
.\" -*- mode: troff; coding: utf-8 -*-
.\" Automatically generated by Pod::Man 5.0102 (Pod::Simple 3.45)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" \*(C` and \*(C' are quotes in nroff, nothing in troff, for use with C<>.
.ie n \{\
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.nr rF 0
.if \n(.g .if rF .nr rF 1
.if (\n(rF:(\n(.g==0)) \{\
.    if \nF \{\
.        de IX
.        tm Index:\\$1\t\\n%\t"\\$2"
..
.        if !\nF==2 \{\
.            nr % 0
.            nr F 2
.        \}
.    \}
.\}
.rr rF
.\" ========================================================================
.\"
.IX Title "ISO-MIRROR 1"
.TH ISO-MIRROR 1 2024-11-30 0.1 "iso-mirror manual"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH NAME
iso\-mirror \-\- mirror set of .iso files for use in multiboot environment like Ventoy
.SH SYNOPSIS
.IX Header "SYNOPSIS"
iso-mirror [\fB\-\-config\fR=\fIfile\fR] [\fB\-\-dry\-run\fR] [\fB\-\-version\fR]
.PP
iso-mirror \fB\-\-help\fR
.SH DESCRIPTION
.IX Header "DESCRIPTION"
\&\fBiso-mirror\fR is
.SH OPTIONS
.IX Header "OPTIONS"
.IP "\fB\-c \-\-config\fR=\fIfile\fR   \- specify location of `config.yaml` file" 1
.IX Item "-c --config=file - specify location of `config.yaml` file"
.PD 0
.IP "\fB\-s \-\-system\fR=\fIname\fR   \- operate only on given system (e.g. ubuntu)" 1
.IX Item "-s --system=name - operate only on given system (e.g. ubuntu)"
.IP "\fB\-n \-\-dryrun\fR        \- check pages and show what would be downloaded, also show what would be deleted when `delete\-obsolete` configuration option is enabled" 1
.IX Item "-n --dryrun - check pages and show what would be downloaded, also show what would be deleted when `delete-obsolete` configuration option is enabled"
.IP "\fB\-d \-\-curldebug\fR     \- enable libcurl debug output" 1
.IX Item "-d --curldebug - enable libcurl debug output"
.PD
.SH "EXIT STATUS"
.IX Header "EXIT STATUS"
In case of errors downloading something \fBiso-mirror\fR will exit with status 1.
In case of fatal errors (like syntax error in configuration file) exit code will be 255.
.SH AUTHOR
.IX Header "AUTHOR"
Marcin Gryszalis <mg@fork.pl>
.SH "SEE ALSO"
.IX Header "SEE ALSO"
https://github.com/marcin\-gryszkalis/iso\-mirror
.SH "COPYRIGHT AND LICENSE"
.IX Header "COPYRIGHT AND LICENSE"
MIT License
.PP
Copyright (c) 2024 Marcin Gryszkalis
.PP
Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:
.PP
The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.
.PP
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.


iso-mirror.pl
#!/usr/bin/perl
# https://github.com/marcin-gryszkalis/iso-mirror
my $VERSION  = '0.1';

use warnings;
use strict;
use feature qw/state signatures say multidimensional/;
use utf8;
use open qw(:std :encoding(UTF-8));

use File::Slurp;
use Data::Dumper;
use FindBin qw($Bin);
use JSON qw(decode_json);
use YAML ();
use Config::INI::Reader;
use Version::Compare;
use Pod::Usage;
use Getopt::Auto;
our %options;


# use List::Util qw/min max first sum sum0 product all any uniq head tail reduce pairs zip mesh/;
# use List::MoreUtils qw/firstidx frequency mode slide minmax/;
# use POSIX qw/ceil floor/;

sub write_file_ok
{
    write_file($_[0], {binmode => ':utf8'}, $_[1]);
}

my $iso = YAML::LoadFile("$Bin/iso.yaml");
my $cfg = YAML::LoadFile("$Bin/cfg.yaml");
print Dumper $iso;


use Term::Emit ":all", {-color => 1};
# Term::Emit::setopts(-fh => *LOG);

emit "Skrawning all xyzons";
print "\nHey, look at me, I'm printed output!\n";
Term::Emit::setopts (-pos => 0);  # Tell where we left the cursor

# emit_prog - progress
emit_fail {-reason => "dead body"};

emit "System parameter updates";
  emit "CLOCK_UTC";
  #...do_something();
  emit_ok;

  emit "NTP Servers";
  #...do_something();
  emit_error;

  emit "DNS Servers";
  #...do_something();
  emit_warn;

  # curl -o "$file" -z "$file" "$uri"

sub usage {
    pod2usage( -verbose => 2 );
    exit 0;
}


my $count;
sub add { $count += shift @ARGV; }
$count += $options{'--inc'};
print "Count: $count\n";

=pod

=head1 NAME

iso-mirror -- mirror set of .iso files for use in multiboot environment like I<Ventoy>

=head1 SYNOPSIS

iso-mirror [B<--config>=I<file>] [B<--dry-run>] [B<--version>]

iso-mirror B<--help>

=head1 DESCRIPTION

B<iso-mirror> is a wrapper


=head2 -c --config=file    specify location of `config.yaml` file

=head2 -n --dry-run        check pages and show what would be downloaded, also show what would be deleted when `delete-obsolete` configuration option is enabled

The integer argument is not checked.

=head1 EXIT STATUS

In case of errors downloading something B<iso-mirror> will exit with status 1.
In case of fatal errors (like syntax error in configuration file) exit code will be 255.

=head1 AUTHOR

Marcin Gryszalis <mg@fork.pl>

=head1 COPYRIGHT AND LICENSE

=head1 SEE ALSO

=cut

# pod2man -c "iso-mirror manual" iso-mirror > iso-mirror.1